backend/src/tasks/lookingGlass.ts

```diff
@@ -0,0 +1,283 @@
+      threatName: string;
+    };
+    left: {
+      collectionIds: string[];
+      classifications: string[];
+      ticScore: number;
+      owners: string[];
+      ref: {
+        type: string;
+        id: number;
+      };
+      name: string;
+      type: string;
+      threatNames: string[];
+      locations: {
+        city: string;
+        country: string;
+        region: string;
+        geoPoint: {
+          long: number;
+          lat: number;
+        };
+        countryName: string;
+        sources: string[];
+        country2Digit: string;
+        lastSeen: number;
+      }[];
+      ipv4: number;
+      asns: number[];
+      threatIds: string[];
+      cidrv4s: string[];
+    };
+    ref: {
+      type: string;
+      right: {
+        type: string;
+        id: string;
+      };
+      left: {
+        type: string;
+        id: string;
+      };
+    };
+  }[];
+}
+
+async function getCollectionForCurrentWorkspace() {
+  const resource =
+    'https://delta.lookingglasscyber.com/api/v1/workspaces/' +
+    process.env.LG_WORKSPACE_NAME +
+    '/collections';
+  return (await got(resource, {
+    headers: { Authorization: 'Bearer ' + process.env.LG_API_KEY }
+  }).json()) as LGCollectionsResponse[];
+}
+
+function validateIPAddress(ipAddress) {
+  return /\b\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\b/.test(ipAddress);
+}
+
+async function getThreatInfo(collectionID) {
+  const resource = 'https://delta.lookingglasscyber.com/api/graph/query';
+  const modifier = {
+    period: 'all',
+    query: [
+      'and',
+      ['=', 'type', ['associated-with']],
+      ['or', ['=', 'right.type', 'threat'], ['=', 'left.type', 'ipv4']],
+      ['=', 'left.collectionIds', collectionID]
+    ],
+    fields: ['firstSeen', 'lastSeen', 'sources', 'right', 'left'],
+    limit: 100000,
+    workspaceIds: []
+  };
+
+  return (got
+    .post(resource, {
+      json: modifier,
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: 'Bearer ' + process.env.LG_API_KEY
+      }
+    })
+    .json() as unknown) as LGThreatResponse[];
+}
+
+async function saveAndPullDomains(response, organizationId, scanId) {
+  const domains: Domain[] = [];
+  const data = response;
+  for (const l of data.results) {
+    if (validateIPAddress(l.left.name)) {
+      const currentDomain = plainToClass(Domain, {
+        name: l.left.name,
+        ip: l.left.name,
+        organization: { id: organizationId },
+        fromRootDomain: '',
+        ipOnly: true,
+        discoveredBy: { id: scanId }
+      });
+      domains.push(currentDomain);
+    } else {
+      const currentDomain = plainToClass(Domain, {
+        name: l.left.name,
+        ip: l.left.name,
+        organization: { id: organizationId },
+        fromRootDomain: '',
+        ipOnly: false,
+        discoveredBy: { id: scanId }
+      });
+      domains.push(currentDomain);
+    }
+  }
+  await saveDomainsToDb(domains);
+
+  await connectToDatabase();
```

Comment:
We can remove this (connectToDatabase is already called in saveDomainsToDb, I think -- it has to be called only once before you do any db operations)
```suggestion
```